frontend/src/reducers/cartReducers.js

```diff
@@ -0,0 +1,33 @@
+import {CART_ADD_ITEM, CART_REMOVE_ITEM} from '../constants/cartConstants'
+
+export const cartReducer = (state = {cartItems:[]},action) =>{
+    switch(action.type){
+        case CART_ADD_ITEM:
+            const item = action.payload
+            const existItem = state.cartItems.find((x) => x.product === item.product)
```

Comment:
try to use good naming convention even in the small loops
`const existItem = state.cartItems.find((cartItem) => cartItem.product === item.product)`

Reply:
Okay i got ur point, thanks